Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/ApplePushNotificationService/ApplePushNotificationServiceConnector.cs

```diff
@@ -78,14 +78,16 @@ private async Task MapResponse(HttpResponseMessage response, PnsRegistration reg
         }
     }
 
-    private static string GetServer(Environment environment)
+    private static string BuildUrl(Environment environment, string handle)
     {
-        return environment switch
+        var baseUrl = environment switch
         {
             Environment.Development => "https://api.sandbox.push.apple.com:443/3/device/",
             Environment.Production => "https://api.push.apple.com:443/3/device/",
             _ => throw new ArgumentOutOfRangeException()
         };
+
+        return $"{baseUrl}{handle}";
```

Comment:
I would move the slash to the concatenation:

```suggestion
            Environment.Development => "https://api.sandbox.push.apple.com:443/3/device",
            Environment.Production => "https://api.push.apple.com:443/3/device",
            _ => throw new ArgumentOutOfRangeException()
        };

        return $"{baseUrl}/{handle}";
```

Looks "cleaner" imo.